File: 9. Palindrome Number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) { //-121
            return false;
        }
        
        int sub = x;
        int reversed = 0;
        while (sub != 0) {
            reversed = reversed * 10 + sub % 10;
            sub /= 10;
        }
        
        return x == reversed;
    }
}

File: Python/9. Palindrome Number.py
class Solution(object):
    def isPalindrome(self, x):
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
    
        reversed_number = 0
        while x > reversed_number:
            reversed_number = reversed_number * 10 + x % 10
            x //= 10
    
        return x == reversed_number or x == reversed_number // 10
    
    
    
    